### PR TITLE
feat: custom subgrounds client for polars

### PR DESCRIPTION
**Pull request by Evan-Kim2028**
Related to this issue - https://github.com/0xPlaygrounds/subgrounds/issues/29

```python
from polars_client import SubgroundsPolars
from subgrounds.subgraph import FieldPath, Subgraph

# from polars_utils.py import *

import polars as pl

sg = SubgroundsPolars()

snx_endpoint = "https://api.thegraph.com/subgraphs/name/synthetix-perps/perps"

snx = sg.load_subgraph(
    url=snx_endpoint,
)

trades_json = sg.query_json(
    [
        # set the first parameter to a larger size to query more rows.
        snx.Query.futuresTrades(
            first=2500,
            orderBy="timestamp",
            orderDirection="desc",
            # where=[{"timestamp_lte": "1694131200"}],  # 1694131200 = 9/8/23
        )
    ]
)
```

### Features
* create a custom subgrounds client to support polars dataframes

### TODO
- [ ] uses custom polars utility functions to help convert from graphql response to polars dataframe. This is currently done, but need to work out importing functions from  `polars_utils.py`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ subgrounds.egg-info/
 
 # apple
 .DS_Store
+.venv*/
```
Comment (0xMochan) [inline .gitignore:18]: We shouldn't need to do this, there should already be a `.gitignore` in your `.venv`.

File: subgrounds/contrib/polars/__init__.py
Comment (0xMochan) [inline subgrounds/contrib/polars/__init__.py:1]: This is important for api purposes, we can refine what we export later.

File: subgrounds/contrib/polars/client.py
Comment (0xMochan) [inline subgrounds/contrib/polars/client.py:1]: This client looks virtually the same as a standard `Subgrounds` client minus the `query_df` things. We'll want to introduce a new `query` or `query_df`, whichever naming you prefer using the utils you added.

The expected interface could look something like this:
```py
>>> from subgrounds.contrib.polars import PolarsSubgrounds
>>> 
>>> sg = PolarsSubgrounds()
>>> subgraph = sg.load_subgraph("...")
>>> 
>>> sg.query(
...     subgraph.Query.transfers(first=100)
... )
# Polars Output
```

This way, you completely interact w/ polars. We can setup a call to pair program if you want some guidance on how we can get here!

File: subgrounds/contrib/polars/utils.py
Comment (0xMochan) [inline subgrounds/contrib/polars/utils.py:1]: Good comments, though you can update some of the structure / casing to match the style internally. Also, there's no need to import `TYPE_CHECKING`.

File: tests/contrib/polars/test_client.py
Comment (0xMochan) [inline tests/contrib/polars/test_client.py:1]: This test file is not structure for `pytest`, it's more like a testing file. We can consider adding this to `if __name__ == "__main__"` in the main polars client file.

In terms of actual tests, if you aren't familiar with pytest, it might be a bit hard to follow how we structure tests in the `tests` directory. We can set a call for this or I can write a mock for how it could look like.

**Review by 0xMochan — CHANGES_REQUESTED** [review]
There's going to be some work before we get to a `PolarsClient`, we can set up a call to go over more about this!

Also, I've renamed some files to match some of our naming conventions and so forth.

File: subgrounds/contrib/polars/client.py
Comment (Evan-Kim2028) [reply subgrounds/contrib/polars/client.py:1]: added a `query_df()` function into the client

File: tests/contrib/polars/test_client.py
Comment (Evan-Kim2028) [reply tests/contrib/polars/test_client.py:1]: correct - `test_client.py` is simply a testing file to ensure that the functionality works. It does not cover any edge cases, bugs, or all pieces of the polars client such as testing any of the polars utils functions.

what kind of pytest tests would you recommend to implement, if any? I see a lot of existing tests for the current sync subgrounds, unsure if any of them would be good to copy.

File: .gitignore (hunk 1 above)
Comment (Evan-Kim2028) [reply .gitignore:18]: there was not so I had to add it in. 

Do you also mean that the `.gitignore` I have should not be getting pushed?

Comment (0xMochan) [reply .gitignore:18]: The `.gitignore` is auto-generating if the `.venv` is created via python's built in `venv` module (which is also what `poetry` uses for generating environments).

**Comment by Evan-Kim2028** [comment]
- refactored function names, comments, and style for utils
- added functional tests to see how the functions work
